h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -59,7 +58,8 @@
     boolean decimal = false;
     int fractionDigits = 0;
     int tokenStart = 0; // used for numeric token to backtrace if not successful
-    int colIdx = 0;
+    int columnCounter = 0;  // index into parsed columns only, exclude skipped columns.
+    int colIdx = 0; // count each actual column in the dataset including the skipped columns
```

Comment:
This should not be named `colIdx` but rather something like `colCnt` ? A minor thing, but when looking at the variable without the comment, it makes me think about a stored ID of an actual column. Is that the purpose ?

Reply:
This is legacy from Tomas (or whoever implement it).  It is column index in the short form.  It indexes into each column including those to be skipped.